### PR TITLE
Update the GPU matrix-free documentation

### PR DESCRIPTION
**Pull request by peterrum**
... while reading the GPU documentation

### CODE REVIEW

**Comment by masterleinad** [comment]
Why is this WIP? Do you want to do more here?

**Comment by peterrum** [comment]
I thought I'll keep it until I have finished the whole documentation...

**Comment by peterrum** [comment]
@masterleinad are you aware that the constructor is missing https://www.dealii.org/developer/doxygen/deal.II/structCUDAWrappers_1_1MatrixFree_1_1AdditionalData.html and all the fields are missing from https://www.dealii.org/developer/doxygen/deal.II/structCUDAWrappers_1_1MatrixFree_1_1Data.html ?

I would like to add some documentation to them in this PR.

File: include/deal.II/matrix_free/cuda_matrix_free.h [hunk 1]
```diff
@@ -142,15 +145,52 @@ namespace CUDAWrappers
...
     struct Data
     {
-      point_type *             q_points;
+      /**
+       * Pointer to the quadrature points.
+       */
+      point_type *q_points;
+
+      /**
+       *Map the position in the local vector to the position in the global
```
Comment (Rombur) [inline include/deal.II/matrix_free/cuda_matrix_free.h]: Missing a blank space

Comment (peterrum) [reply include/deal.II/matrix_free/cuda_matrix_free.h]: Done! Ready!

**Pull request merged**